src/main/java/build/buf/protovalidate/internal/evaluator/ConstraintResolver.java

```diff
@@ -32,18 +34,24 @@ class ConstraintResolver {
    * @param desc the message descriptor.
    * @return the resolved {@link MessageConstraints}.
    */
-  MessageConstraints resolveMessageConstraints(Descriptor desc) {
+  MessageConstraints resolveMessageConstraints(Descriptor desc)
+      throws InvalidProtocolBufferException {
     DescriptorProtos.MessageOptions options = desc.getOptions();
     if (!options.hasExtension(ValidateProto.message)) {
       return MessageConstraints.getDefaultInstance();
     }
-
-    MessageConstraints constraints = options.getExtension(ValidateProto.message);
```

Comment:
This is the location where users were seeing:

```
mergeFrom(Message) can only merge messages of the same type.
java.lang.IllegalArgumentException: mergeFrom(Message) can only merge messages of the same type.
```

This occurred when the generated .java files from protoc referenced protovalidate extensions in another location (i.e. `build.buf.gen.buf.validate.*` for remote packages) while we tried to read them as `build.buf.validate.*` classes.

We should still resolve this in remote packages to avoid having duplicate classes generated to different Java packages on the classpath, but this at least avoids the exception and also has the side effect of working for users that use `buf generate` with managed mode.